doc/nrf/ug_creating_matter_accessory.rst

```diff
@@ -0,0 +1,274 @@
+                        break;
+                case AppEvent::SensorDeactivate:
+                        SensorDeactivateHandler();
+                        break;
+                case AppEvent::SensorMeasure:
+                        SensorMeasureHandler();
+                        break;
+                default:
+                        LOG_INF("Unknown event received");
+                        break;
+                }
+        }
+        
+Create a callback for sensor activation/deactivation
+****************************************************
+
+Up until now events ``SensorActivate`` and ``SensorDeactivate`` were never added to the event queue.   
+This is because attribute of the OnOff cluster is never changed from within ``AppTask``. 
+It can only be changed externally, for example through Matter controller.
+
+
+However, sensor application needs to know how this attribute is changing, so there is a need to implement a callback function to post one of these events every time OnOff attribute changes.
+To see a list of callback functions that are customizable one can look into :file:`src/gen/callback-stub.cpp` and look for functions with ``__attribute__((weak))``.      
+It is sufficient just to implement ``emberAfPostAttributeChangeCallback`` (read the description in :file:`src/gen/callback-stub.cpp`).
+
+
+In order to do it, create a new file, for example :file:`src/zcl_callbacks.cpp` and implement a callback inside it.
+A simple exemplary implementation might look like this:     
+
+.. code-block:: C++
+
+        #include "app_task.h"
+        #include <app/util/af.h>
+
+        using namespace ::chip;
+
+        void emberAfPostAttributeChangeCallback(EndpointId endpoint, ClusterId clusterId, AttributeId attributeId, uint8_t mask,
+                                                uint16_t manufacturerCode, uint8_t type, uint16_t size, uint8_t *value)
+        {
+                GetAppTask().PostEvent(AppEvent(*value ? AppEvent::SensorActivate : AppEvent::SensorDeactivate));
+        }
+                
+
+
+Add new targets to CMakelists
+*****************************
+
+Now, to allow a proper build, update CMakelists.txt file.
+
+Add the following targets:
+
+#. :file:`$MATTER_ROOT/src/app/clusters/on-off-server/on-off-server.cpp` - for OnOff cluster callbacks.
+#. :file:`$MATTER_ROOT/src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp` - for TemperatureMeasurement cluster callbacks.
+#. :file:`src/zcl_callbacks.cpp` - to include our callback implementation.
+
+Don't forget to include:
+
+.. code-block::
+        
+        #include <app/clusters/on-off-server/on-off-server.h>
+        #include <app/clusters/temperature-measurement-server/temperature-measurement-server.h>
+
+in :file:`src/app_task.cpp` file.
+
+Testing sensor application
+**************************
+
+In order to check if sensor device works correctly complete following steps:
+
+#. Commision device using Matter controller.
```

Comment:
Maybe I don't know the full context of this guide, but for me some reference to commissioning guide is missing. I can imagine that somebody will get here from some weird location (e.g. Google browser) and will be a bit confused how to commission the device and invoke commands to the controller, so please consider adding it in case you find it beneficial too.

Reply:
I think we can refer to the Matter Template here as well. The commissioning should be no different for the new sample.